ttnn/cpp/ttnn/operations/conv/conv_transpose2d/conv_transpose2d.cpp

```diff
@@ -0,0 +1,350 @@
+            .dilation_hw = {dilation[0], dilation[1]},
+            .is_transpose = true
+        };
+
+        uint32_t output_height = (input_height - 1) * stride[0] - 2 * padding[0] + dilation[0] * (kernel_size[0] - 1) + output_padding[0] + 1;
+        uint32_t output_width  = (input_width  - 1) * stride[1] - 2 * padding[1] + dilation[1] * (kernel_size[1] - 1) + output_padding[1] + 1;
+
+        //Dimensions of Input to Conv u_op
+        uint32_t full_input_height = output_height + dilation[0] * (kernel_size[0] - 1);
+        uint32_t full_input_width  =  output_width + dilation[1] * (kernel_size[1] - 1);
+
+        //Size of input after adding interleaved 0s.
+        uint32_t strided_input_height = (input_height - 1) * stride[0] + 1;
+        uint32_t strided_input_width  = (input_width -  1) * stride[1] + 1;
+
+        uint32_t input_pad_top  = (full_input_height - strided_input_height)/2;
+        uint32_t input_pad_bottom = full_input_height - strided_input_height - input_pad_top;
+
+        uint32_t input_pad_left = (full_input_width - strided_input_width)/2;
+        uint32_t input_pad_right = full_input_width - strided_input_width - input_pad_left;
+
+        log_debug(LogOp, "Input : {}x{}", input_height, input_width);
+        log_debug(LogOp, "Output : {}x{}", output_height, output_width);
+
+        log_debug(LogOp, "Conv Op Input : {}x{}", full_input_height, full_input_width);
+        log_debug(LogOp, "Strided Input : {}x{}", strided_input_height, strided_input_width);
+
+        log_debug(LogOp, "Padding : ({},{}) ({},{})", input_pad_top, input_pad_bottom, input_pad_left, input_pad_right);
+
+
+        DeviceComputeKernelConfig compute_kernel_config;
+        switch (device->arch()) {
+            case tt::ARCH::WORMHOLE_B0:
+                compute_kernel_config = WormholeComputeKernelConfig(
+                    {.math_fidelity = conv_config.math_fidelity,
+                    .math_approx_mode = conv_config.math_approx_mode_enabled,
+                    .fp32_dest_acc_en = conv_config.fp32_dest_acc_enabled,
+                    .packer_l1_acc = conv_config.packer_l1_accum_enabled});
+                break;
+
+            case tt::ARCH::GRAYSKULL:
+                compute_kernel_config = GrayskullComputeKernelConfig(
+                    {.math_fidelity = conv_config.math_fidelity, .math_approx_mode = conv_config.math_approx_mode_enabled});
+                break;
+
+            case tt::ARCH::BLACKHOLE:
+                compute_kernel_config = BlackholeComputeKernelConfig(
+                    {.math_fidelity = conv_config.math_fidelity,
+                    .math_approx_mode = conv_config.math_approx_mode_enabled,
+                    .fp32_dest_acc_en = conv_config.fp32_dest_acc_enabled,
+                    .packer_l1_acc = conv_config.packer_l1_accum_enabled});
+                break;
+
+            default:
+                TT_THROW("Invalid Device Arch, Got {}",device->arch());
+        }
```

Comment:
Extract into a free function (in an anonymous namespace). Something like this:

```suggestion
    DeviceComputeKernelConfig compute_kernel_config = get_compute_kernel_config(conv_config);
```

Proper anonymous namespace will have to look like this to work safely with unity builds
```
namespace {
namespace CMAKE_UNIQUE_NAMESPACE {
.... your free function
}
}
```

Reply:
[This PR ](https://github.com/tenstorrent/tt-metal/pull/14210) removes the compute kernel configs from Conv2dConfig. When those changes are included, this code will be removed.